var/spack/repos/builtin/packages/flux-core/package.py

```diff
@@ -56,25 +69,59 @@ class Flux(AutotoolsPackage):
     depends_on("automake", type='build', when='@master')
     depends_on("libtool", type='build', when='@master')
 
-    def setup():
+    def setup(self):
         pass
 
     @when('@master')
     def setup(self):
         # Allow git-describe to get last tag so flux-version works:
         git = which('git')
-        git('pull', '--depth=50', '--tags')
+        git('fetch', '--unshallow')
 
     def autoreconf(self, spec, prefix):
         self.setup()
-        if os.path.exists('autogen.sh'):
+        if not os.path.exists('configure'):
             # Bootstrap with autotools
             bash = which('bash')
             bash('./autogen.sh')
 
+    @property
+    def lua_version(self):
+        return self.spec['lua'].version.up_to(2)
+
+    @property
+    def lua_share_dir(self):
+        return os.path.join('share', 'lua', str(self.lua_version))
+
+    @property
+    def lua_lib_dir(self):
+        return os.path.join('lib', 'lua', str(self.lua_version))
+
     def setup_environment(self, spack_env, run_env):
         #  Ensure ./fluxometer.lua can be found during flux's make check
         spack_env.append_path('LUA_PATH', './?.lua', separator=';')
 
+        run_env.prepend_path(
+            'LUA_PATH',
```

Comment:
Rather than doing this, would it be possible to just `extend('lua')` & `extend('python')`?

Reply:
Try it, if it works that would be preferable for sure.  IIRC it used to be limited to one extension per package.

Reply:
Correct, see #987 😢

Reply:
Ok. I subscribed to #987.  Once it gets closed, I'll switch flux over to extending both `python` and `lua` rather than manually adding the paths.